src/components/CharacterGallery.js

```diff
@@ -0,0 +1,64 @@
+import React from 'react';
+import { withRouter } from 'react-router-dom';
+import { getCharacters } from '../util/swapi';
+import CharacterItem from './CharacterItem';
+import './CharacterGallery.css';
+
+class CharacterGallery extends React.Component {
+    constructor(props) {
+        super(props);
+        this.state = {
```

Comment:
This component is a perfect candidate to refactor it and make it a FC with `useState` and `useHistory` hooks.

Reply:
CharacterGallery.js is now a functional component.